### DISCUSSION TITLE
Run renovate locally without using any platform

### DISCUSSION THREAD

**Discussion by bilak** [renovatebot/renovate#24846] [Suggest an Idea]
### What would you like Renovate to be able to do?

While renovate bot is running correctly in platforms such gitlab/gihub it's sometimes required to test changes locally (e.g. sometimes it's much faster, sometimes one does not have privileges to change some configuration).

To do so it would be nice to leverage the [docker images](https://github.com/renovatebot/docker-renovate-full) where it could be possible to:
* include local repositories as volume
* platform could be set to something like `local`
* only branches are created when new versions are found

```js
module.exports = {
    endpoint: 'file:///git-repositories',
    platform: 'local',
    onboardingConfig: {
        extends: ['config:base'],
    },
    repositories: [
        {
            repository: 'repo1,
            packageRules: [
                {
                    description: 'Disable MAJOR update types',
                    matchUpdateTypes: ['major'],
                    enabled: false,
                },
            ],
        },
    ],
}
```

so if there is a parent directory called `work` containing subdirectories (e.g. `repo1`, `repo2`) with git project one can run following docker command:

`$ docker run --rm -it -v $PWD/config.js:/usr/src/app/config.js -v /path/to/work:/git-repositories -e LOG_LEVEL=debug renovate/renovate`

### If you have any ideas on how this should be implemented, please tell us here.

not an javascript dev :(

### Is this a feature you are interested in implementing yourself?

No

**Comment by nabeelsaabna** [comment]
Hi @bilak 
would [dry-run](https://docs.renovatebot.com/self-hosted-configuration/#dryrun) work ?

**Comment by bilak** [comment]
@nabeelsaabna can I reference local repository with dry-run?

**Comment by nabeelsaabna** [comment]
no, Renovate runs only with platform.
the dry-run means the remote repository is read-only

**Comment by bilak** [comment]
@nabeelsaabna sorry for delay...this will still need some changes to be pushed to remote repository.  I'm struggling with gradle setup so I don't want to mess with it and affect that other developers. It would be really useful to have an option to execute renovate against local repo.

**Comment by viceice** [comment]
you should use a fork to test against

**Comment by bilak** [comment]
@viceice still not very practical. I have to disable ci/cd and don't know what other stuff. Besides that if I'm not a gitlab admin I will have hard times to configure the bot correctly.

**Comment by childnode** [comment]
I expect this is outdated, because there is the platform local available https://docs.renovatebot.com/modules/platform/local/
since [35.76.0](https://github.com/renovatebot/renovate/releases/tag/35.76.0) and documented as of [35.77.0 back in 2023](https://github.com/renovatebot/renovate/releases/tag/35.77.0) << https://github.com/renovatebot/renovate/pull/22010

**Reply by astellingwerf** [reply]
The problem with the local platform is that it doesn't properly support preset resolution. It shouldn't be too hard to resolve them relative to a folder on the filesystem, right?
